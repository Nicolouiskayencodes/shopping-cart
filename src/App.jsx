import { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import './App.css'
import Header from './components/header'
import Home from './components/home'
import Shopping from './components/shopping'

function App() {
  const {page} = useParams()
  const [items, setItems] = useState(null)
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(true);
  const [cart, setCart] = useState([]);
  useEffect(()=>{
    fetch('https://fakestoreapi.com/products?limit=20', {mode: 'cors'})
      .then(response=>{ 
        if (response.status >= 400) {
          throw new Error("server error");
        }
        return response.json()})
      .then(response=>setItems(response))
      .catch((error) => setError(error))
      .finally(()=> setLoading(false))
  }, [])
  function addCart(item){
    setCart([...cart, item]);
  }
   console.log(cart)

  return (
    <>
      <Header />
      {page === 'shopping' ? (
        (loading && <p>Loading...</p>),
        (error && <p>A network error was encountered</p>),
        (items && <Shopping props={items} add={addCart}/>)
      ): (
        <Home />
      )}
   </>
  )
}

export default App
