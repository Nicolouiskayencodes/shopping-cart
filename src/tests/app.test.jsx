import { describe, it, expect } from "vitest";
import { BrowserRouter } from "react-router-dom";
import { render, } from '@testing-library/react';
import App from '../App';

describe("App", () => {
  it('App snapshot', () => {
    const { container } = render(<BrowserRouter><App /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  })
});