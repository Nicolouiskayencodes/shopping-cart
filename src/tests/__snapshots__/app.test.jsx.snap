// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > App snapshot 1`] = `
<div>
  <div
    class="_header_5800ce"
  >
    <h1
      class="_title_5800ce"
    >
      InterPorium
    </h1>
    <div
      class="_nav_5800ce"
    >
      <a
        class="_active_5800ce"
        href="/"
      >
        Home
      </a>
      <a
        href="/shopping"
      >
        Shop
      </a>
      <a
        href="/cart"
      >
        <img
          class="_img_5800ce"
          src="/src/assets/cart-variant.svg"
        />
         
      </a>
    </div>
  </div>
  <div
    class="_container_3f2dd1"
  >
    <img
      alt=""
      class="_img_3f2dd1"
      src="/src/assets/checkout.jpg"
    />
    <div
      class="_right_3f2dd1"
    >
      <p
        class="_info_3f2dd1"
      >
        Welcome to the InterPorium! We are a sell a large assortment of high quality and affordable goods. We source our products from community oriented and environmentally friendly small businesses worldwide! We are dedicated to positive customer experiences and linking buyers and sellers across the globe. If you are interested in getting in contact, email us at interporium@fakebusinessinquiries.net. We hope you like what you see!
      </p>
      <p
        class="_extra_3f2dd1"
      >
        Photo by 
        <a
          href="https://unsplash.com/@blakewisz?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash"
        >
          Blake Wisz
        </a>
         on 
        <a
          href="https://unsplash.com/photos/person-holding-smartphone-beside-tablet-computer-Xn5FbEM9564?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash"
        >
          Unsplash
        </a>
      </p>
      <div
        class="_extra_3f2dd1"
      >
        InterPoriumâ„¢ nicolouiskayencodes 2024
      </div>
    </div>
  </div>
</div>
`;
